odoo/addons/base/models/ir_translation.py

```diff
@@ -922,3 +922,44 @@ def get_web_translations_hash(self, mods, lang):
             'multi_lang': len(self.env['res.lang'].sudo().get_installed()) > 1,
         }
         return hashlib.sha1(json.dumps(translation_cache, sort_keys=True).encode()).hexdigest()
+
+    @api.model
+    def _guess_language_of_terms(self, terms, types=None):
+        """Guess the language of the terms based on the translations stored in the database.
+
+        Note that the search relies on some non indexed fields.
+        :param tuple(str) terms: terms used in the application for which we want to guess the language
+        :param set types: if set, limit the search to the given types (increase performance)
+        :return tuple(str, float):
+            - best_candidate (str): the best language (code) candidate or None if none (ex.: fr_BE)
+            - best_candidate_term_ratio (float): ratio of term found in the database for the best_candidate language
+        """
+        params = dict(terms=tuple([t.lower().strip() for t in terms]), main_lang='en_US')
```

Comment:
We could probably use unaccent if activated (see ``get_unaccent_wrapper(self._cr)``). This could probably help having better results.

Reply:
Thanks for the tip. I've updated the code with that method. Note that:
- As we compare the terms from the header of the imported file and the terms in the database, we must apply the same transformation to both of them. As the "unaccent" is not available in python, we use a pre-query to normalize the term to be searched.
- As the module is not always present in the test environment, there are no test about it but I have tested that it works with the module installed locally.